app/lib/report_helper.rb

```diff
@@ -427,10 +427,26 @@ def top_taxons_details(samples, background_id, num_results, sort_by_key, only_sp
       taxon_counts = res["taxon_counts"]
       sample_id = pr.sample_id
       tax_2d = convert_2d(taxon_counts)
+      if only_species
+        puts "ONLY SPECIES"
+        puts "TAX LEVEL SPECIES: " + TaxonCount::TAX_LEVEL_SPECIES.to_s
+        tax_2d.keep_if { |_tax_id, tax_info| tax_info['tax_level'] = TaxonCount::TAX_LEVEL_SPECIES }
+      else
+        puts "NOT ONLY SPECIES"
+        puts "TAX LEVEL GENUS: " + TaxonCount::TAX_LEVEL_GENUS.to_s
+        puts "BEFORE"
+        puts tax_2d
+        tax_2d.keep_if { |_tax_id, tax_info| tax_info['tax_level'] = TaxonCount::TAX_LEVEL_GENUS }
+        puts "AFTER"
+        puts tax_2d
+      end
+      tax_2d = validate_names!(tax_2d)
```

Comment:
so this line needs to be above the above filter, not below

Reply:
and you are right why not use the same cleanup_all function for all the cleanup, instead of unfolding it --- may be there was a good reason for that, but probably the reason was not good :P)